src/frame.h

```diff
@@ -57,7 +57,7 @@ struct frame_additional_data : frame_header
 {
     uint32_t metadata_size = 0;
     bool fisheye_ae_mode = false;  // TODO: remove in future release
-    std::array< uint8_t, MAX_META_DATA_SIZE > metadata_blob;
+    std::array< uint8_t, RS2_FRAME_METADATA_ACTUAL_COUNT * sizeof( rs2_metadata_type ) > metadata_blob;
```

Comment:
Why not use an array of rs2_metadata_type?

Reply:
Because that's just one usage for software sensors. Most time it's binary data.